packages/clay-data-provider/src/useResource.ts

```diff
@@ -170,6 +171,11 @@ const useResource = ({
 	const doFetch = (retryAttempts = 0) => {
 		let promise: Promise<any>;
 
+		warning(
+			typeof link === 'string',
+			'DataProvider: The behavior of the `link` accepting a function has been deprecated in favor of the `fetcher` API. \n More details: https://clayui.com/docs/components/data-provider.html#data-fetching'
+		);
+
```

Comment:
Do we really need to deprecate this? The cost of maintaining it seems to be very close to zero, and if we spew a warning into the console we are really just creating churn for little benefit.

Reply:
As I said in the comment below is that using the link to pass an async function brings some problems to the Data Provider, so the idea is to add warnings so that we can remove this behavior in version 4.0 and just stay with the `fetcher` API .